app/lib/katello/util/http_proxy.rb

```diff
@@ -2,7 +2,7 @@ module Katello
   module Util
     module HttpProxy
       def proxy_uri
-        URI("#{proxy_scheme}://#{proxy_user_info}@#{proxy_host}:#{proxy_port}").to_s if proxy_host
+        CGI.escape("#{proxy_scheme}://#{proxy_user_info}@#{proxy_host}:#{proxy_port}").to_s if proxy_host
```

Comment:
Does this actually fix the problem? 
```
> CGI.escape("http://foo:bar@example.com:3000")
=> "http%3A%2F%2Ffoo%3Abar%40example.com%3A3000"
```

I don't think that will work

Reply:
I have reviewed this again, and this doesn't work. Feel free to remove this PR from the bug.

Thanks